### DISCUSSION TITLE
Retrieving count of related models from accessor

### DISCUSSION THREAD

**Discussion by daniser** [CodeDredd/pinia-orm#857] [Q&A] [answered]
Hello! Thank you for wonderful product!

My question is:
How can I get related model count from within model's accessor?

For example:
```ts
export default class Operator extends Model {
    @HasMany(() => Ticket, "operator_id") declare tickets: Ticket[];

    get ticket_count() {
        return this.tickets?.length;
    }
}

```
`operator.ticket_count` returns undefined.

P.S. If I understand correctly, models are dumb and cannot be used for querying the data.
I want to sort operators by the number of associated tickets. This doesn't work:
```ts
const sortedOperators = computed(() => {
    return operatorRepo.value
        .with("tickets")
        .orderBy("ticket_count", "desc")
        .get();
});
```
But this works:
```ts
const sortedOperators = computed(() => {
    return useCollect(
        operatorRepo.value.with("tickets").get()
    ).sortBy([["ticket_count", "desc"]]);
});
```
Am I doing it the right way?

**Comment by CodeDredd** [comment, marked as answer]
Yes you are doing it the right way. Also in `vuex-orn` you never could access relations because they are loaded to late. Maybe i try for v2. make this possible, but first i have to look into the code what this would mean.
